tests/join_test.py

```diff
@@ -129,7 +130,7 @@ def test_lookup(self):
         res3 = ak.lookup(keys[::-1], values[::-1], args, fillvalue=-1)
         self.assertTrue((res3.to_ndarray() == ans).all())
         # Non-unique keys should raise error
-        with self.assertRaises(ak.NonUniqueError):
+        with pytest.warns(UserWarning):
```

Comment:
in other places in our tests, we use
```python
with self.assertWarns(UserWarning):
```
I'm not sure if one is better than the other, but we should prob be consistent

Reply:
Ah, thanks! I was looking for a `self` method related to warnings, but couldn't find it.